reproman/interface/docker.shim

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/perl
+#emacs: -*- mode: shell-script; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: t -*-
+#ex: set sts=4 ts=4 sw=4 noet:
+#
+#
+# This script is a shim for the docker executable. The shim intercepts
+# calls to the docker executable, runs a image id discovery process, and
+# then proceeds with the original call to the original docker executable.
+#
+require 5.002;
+
+# Print debugging if docker debug switch is set
+my $DEBUG = 0;
+foreach(@ARGV) {
+    $DEBUG = 1 if ($_ eq '--debug' or $_ eq '-D');
+}
```

Comment:
Hmm, because we're going to go through the arguments later to look for "run", I think we should just do so now and

  * look for --debug/-D only in the options before "run"
  * warn if there are any other options before "run" because AFAICS we are currently ignoring them
  * die if "run" is not found (currently we continue with start_index as an undefined variable)

Reply:
For the second bullet, no options are ignored for the running of the command after we are done gathering the info we need:

`$cmd = "${original_docker_exec} " . join(" ", @ARGV);`

We do ignore the options when we are figuring out the container id because they are not consequential for what we are doing for that run.

Reply:
Indeed.  Thanks for the correction.